src/script/ismine.cpp

```diff
@@ -146,23 +144,27 @@ isminetype IsMineInner(const CKeyStore& keystore, const CScript& scriptPubKey, b
             }
         }
         if (HaveKeys(keys, keystore)) {
-            return ISMINE_SPENDABLE;
+            Update(ret, ISMINE_SPENDABLE);
         }
         break;
     }
     }
 
-    if (keystore.HaveWatchOnly(scriptPubKey)) {
+    if (ret == ISMINE_NO && keystore.HaveWatchOnly(scriptPubKey)) {
```

Comment:
commit: Simplify IsMine logic

I get that it's an optimization, but it still strikes me as odd that everywhere else in the function uses `Update` to modify the `ret` value, while this does a check and return.